pkg/routes/images_test.go

```diff
@@ -1732,3 +1735,144 @@ var _ = Describe("Images Route Tests", func() {
 		})
 	})
 })
+
+var _ = Describe("Images Route Integration Tests", func() {
+
+	var ctrl *gomock.Controller
+	var router chi.Router
+	var mockImageBuilder *mock_imagebuilder.MockClientInterface
+	var edgeAPIServices *dependencies.EdgeAPIServices
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		ctx := context.Background()
+		log := log.NewEntry(log.StandardLogger())
+
+		mockImageBuilder = mock_imagebuilder.NewMockClientInterface(ctrl)
+
+		imageService := &services.ImageService{
+			ImageBuilder: mockImageBuilder,
+			RepoBuilder:  services.NewRepoBuilder(ctx, log),
+			RepoService:  services.NewRepoService(ctx, log),
+			Service:      services.NewService(ctx, log),
+		}
+
+		edgeAPIServices = &dependencies.EdgeAPIServices{
+			ImageService: imageService,
+			Log:          log,
+		}
+
+		router = chi.NewRouter()
+		router.Use(func(next http.Handler) http.Handler {
+			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				ctx := dependencies.ContextWithServices(r.Context(), edgeAPIServices)
+				next.ServeHTTP(w, r.WithContext(ctx))
+			})
+		})
+		router.Route("/images", MakeImagesRouter)
+
+		mockImageBuilder.EXPECT().GetCommitStatus(gomock.Any()).DoAndReturn(func(image *models.Image) (*models.Image, error) {
+			return image, nil
+		}).AnyTimes()
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Context("POST /images/{imageId}/update", func() {
+		When("using the latest image with valid information", func() {
+			It("should create image update successfully", func() {
+				repoURL := faker.URL()
+				image, imageSet := seeder.Images().WithRepoURL(repoURL).Create()
+
+				updateImage := models.Image{
+					Name:         image.Name,
+					Distribution: "rhel-91",
+					OutputTypes:  []string{models.ImageTypeCommit},
+					Commit: &models.Commit{Arch: "x86_64",
+						InstalledPackages: []models.InstalledPackage{
+							{Name: "vim"},
+						},
+					},
+				}
+
+				composeImage := updateImage
+				composeImage.Commit.OSTreeParentCommit = repoURL
+				composeImage.Commit.OSTreeRef = "rhel/9/x86_64/edge"
+				composeImage.Commit.OSTreeParentRef = "rhel/9/x86_64/edge"
+				composeImage.OrgID = common.DefaultOrgID
+				composeImage.Account = common.DefaultAccount
+				composeImage.ImageSetID = &imageSet.ID
+
+				var body bytes.Buffer
+				err := json.NewEncoder(&body).Encode(&updateImage)
+				Expect(err).ToNot(HaveOccurred())
+
+				req, err := http.NewRequest("POST", fmt.Sprintf("/images/%d/update", image.ID), &body)
+				Expect(err).ToNot(HaveOccurred())
+
+				mockImageBuilder.EXPECT().ComposeCommit(gomock.Eq(&composeImage)).DoAndReturn(func(image *models.Image) (*models.Image, error) {
```

Comment:
```suggestion
				mockImageBuilder.EXPECT().ComposeCommit(gomock.Eq(&composeImage)).Return(image, nil)
```
we do not need the function here

Reply:
hey @ldjebran thank you for your suggestion
for this one, it's required because by using `Return` only the main code loses the pointer reference and causes panic
with `DoAndReturn` is able to get the pointer and just return it

I double-check with your suggestion and it's causing panic

Reply:
and with the pointer 

```go
mockImageBuilder.EXPECT().ComposeCommit(gomock.Eq(&composeImage)).Return(&image, nil)
```

Reply:
after an offline  discussion and debugging session we came on that we need this function, and we need to return the image received by ComposeCommit that we cannot control from our test.
resolving question